### PR TITLE
feat: migrate name

### PR DESCRIPTION
**Pull request by MohdImran001**
Part of #72.
Test with `npm run build && npx mocha test/name.unit.js`

### CODE REVIEW

File: src/name.ts [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+      }
+      this[name] = this[name].bind(this);
+    }
+  }
+
+  /**
+   * firstName
+   *
+   * @method faker.name.firstName
+   * @param {mixed} gender
```
Comment (Shinigami92) [inline src/name.ts:18]: Please remove the param JSDoc types

**Review by Shinigami92 — CHANGES_REQUESTED** [review]

File: src/name.ts [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+        } else if (gender.toLowerCase() === 'female') {
+          gender = 1;
+        }
+      }
+
+      if (typeof gender !== 'number') {
+        if (typeof this.faker.definitions.name.first_name === 'undefined') {
+          gender = this.faker.datatype.number(1);
+        } else {
+          //Fall back to non-gendered names if they exist and gender wasn't specified
```
Comment (Shinigami92) [inline src/name.ts:41]: Would be cool if we already format a bit such comments

```suggestion
          // Fall back to non-gendered names if they exist and gender wasn't specified
```

File: src/name.ts [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+      suffix: string = '';
+
+    // in particular locales first and last names split by gender,
+    // thus we keep consistency by passing 0 as male and 1 as female
+
+    if (typeof gender !== 'number') {
+      gender = this.faker.datatype.number(1);
+    }
+
+    firstName = firstName || this.faker.name.firstName(gender);
```
Comment (Shinigami92) [inline src/name.ts:152]: I already improved such code with `||=` in the other PRs

File: src/name.ts (hunk 1 above)
Comment (MohdImran001) [reply src/name.ts:18]: we should keep only `@method faker.name.firstName` right?

Comment (Shinigami92) [reply src/name.ts:18]: param is actually okay, we can add doc messages/descriptions later

File: src/name.ts [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+      }
+      this[name] = this[name].bind(this);
+    }
+  }
+
+  /**
+   * firstName
+   *
+   * @method faker.name.firstName
+   * @param  gender
```
Comment (Shinigami92) [inline src/name.ts:18]:
```suggestion
   * @param gender
```

File: src/name.ts [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+
+    // in particular locales first and last names split by gender,
+    // thus we keep consistency by passing 0 as male and 1 as female
+
+    if (typeof gender !== 'number') {
+      gender = this.faker.datatype.number(1);
+    }
+
+    firstName ||= this.faker.name.firstName(gender);
+    lastName ||= lastName || this.faker.name.lastName(gender);
```
Comment (Shinigami92) [inline src/name.ts:153]:
```suggestion
    lastName ||= this.faker.name.lastName(gender);
```

**Review by Shinigami92 — APPROVED** [review]

**Review by prisis — APPROVED** [review]

File: src/name.ts [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+    // thus we keep consistency by passing 0 as male and 1 as female
+
+    if (typeof gender !== 'number') {
+      gender = this.faker.datatype.number(1);
+    }
+
+    firstName ||= this.faker.name.firstName(gender);
+    lastName ||= this.faker.name.lastName(gender);
+
+    switch (r) {
```
Comment (prisis) [inline src/name.ts:155]: Just a note, this should be later transformed to a if

**Review by prisis — APPROVED** [review]

**Pull request merged**